api/server/v1alpha1/device_types.go

```diff
@@ -128,9 +131,36 @@ func (c *Device) SetReclaimPolicy(p runtimev1alpha1.ReclaimPolicy) {
 
 // +kubebuilder:object:root=true
 
-// DeviceList contains a list of Device
+// DeviceList contains a list of Devices
 type DeviceList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Device `json:"items"`
 }
+
+// DeviceClassSpecTemplate ...
+type DeviceClassSpecTemplate struct {
+	runtimev1alpha1.NonPortableClassSpecTemplate `json:",inline"`
+	DeviceSpec                                   `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// DeviceClass ...
+type DeviceClass struct {
```

Comment:
The `DeviceClass` will need required methods like those generated here (https://github.com/crossplaneio/stack-gcp/blob/master/apis/cache/v1alpha2/zz_generated.nonportableclass.go) in order to satisfy the `NonPortableClass` interface. We have a library to do this now and I would be happy to make the required updates after this PR is merged if that is easier.

Reply:
If you could that I'd be grateful.

Reply:
Sure thing! Would be happy to

Reply:
@hasheddan Do you need any help with the updates you mentioned above?

Reply:
@jasmingacic I believe I implemented these updates in #2. We have a few changes coming this week that will also require updates that I would be happy to implement :) Also, do you know if it is going to be possible to get this repo name changed back to `stack-packet`?